appinventor/components/src/com/google/appinventor/components/runtime/util/MediaUtil.java

```diff
@@ -504,6 +509,38 @@ private static Bitmap decodeStream(InputStream is, Rect outPadding, BitmapFactor
     // doesn't skip the requested number of bytes.
     return BitmapFactory.decodeStream(new FlushedInputStream(is), outPadding, opts);
   }
+  
+  /**
+   * Loads the image specified by path and returns a Drawable.
+   * (converted into NinepatchDrawable if it can)
+   *
+   * <p/>If mediaPath is null or empty, null is returned.
+   *
+   * @param form the Form
+   * @param path the path to the media
+   * @return a Drawable or null
+   * @author ColinTree (502470184@qq.com)
+   *
+   */
+  public static Drawable getNinePatchDrawable(Form form, String path) throws IOException {
+    if(path.length()==0 || path==null){
+      return null;
+    }
+    InputStream is=MediaUtil.openMedia(form, path);
+    Bitmap bitmap=BitmapFactory.decodeStream(is);
+    BitmapType type=BitmapType.determineBitmapType(bitmap);
+    switch(type){
+      case NULL:
+        return null;
+      case NinePatch:
+        return new NinePatchDrawable(form.getResources(), bitmap, bitmap.getNinePatchChunk(), new Rect(), null);
+      case RawNinePatch:
+        return NinePatchChunk.create9PatchDrawable(form, bitmap, null);
+      //case PlainImage:
+      default:
+        return new BitmapDrawable(form.getResources(), bitmap);
```

Comment:
It would be better to call through to `getBitmapDrawable` here rather than calling `new` directly..

Reply:
@ewpatton But getBitmapDrawable is based on the path to the image, and i have already got a bitmap here.